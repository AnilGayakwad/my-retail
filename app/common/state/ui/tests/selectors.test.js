import { fromJS } from 'immutable';
import makeSelectUiState, {
  makeSelectUiForProductDetailsPageState,
} from '../selectors';

const selector = makeSelectUiState();

describe('ui selectors', () => {
  const state = fromJS({
    ui: {
      newPage: {
        test: 'newPageObj',
      },
      productDetailsPage: {
        test: 'pass',
      },
    },
  });

  it('should return Ui as an object', () => {
    expect(selector(state)).toMatchSnapshot();
  });

  describe('makeSelectUiForProductDetailsPageState', () => {
    it('should return ProductDetails as an object', () => {
      expect(makeSelectUiForProductDetailsPageState()(state)).toEqual({ test: 'pass' });
    });
  });
});
