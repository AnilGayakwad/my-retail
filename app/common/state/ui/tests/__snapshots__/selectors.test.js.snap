// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui selectors should return Ui as an object 1`] = `
Object {
  "newPage": Object {
    "test": "newPageObj",
  },
  "productDetailsPage": Object {
    "test": "pass",
  },
}
`;
